hangupsbot/handlers.py

```diff
@@ -67,8 +67,12 @@ def handle_chat_message(self, event):
             event.print_debug()
 
         if event.text:
+            if event.user.is_self:
+                event.from_bot = True
+            else:
+                event.from_bot = False
             yield from self.run_pluggable_omnibus("allmessages", self.bot, event, command)
-            if not event.user.is_self:
+            if not event.from_bot:
```

Comment:
It might, I will perform a code audit of the API module to determine whether it breaks or not, quite possibly may have to integrate API + reprocessor if the functionality overlaps